static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -0,0 +1,1033 @@
+import {TabList} from 'sentry/components/tabs/tabList';
+import {
+  IconCheckmark,
+  IconCircle,
+  IconMegaphone,
+  IconOpen,
+  IconQuestion,
+  IconRefresh,
+  IconWarning,
+} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+interface SourceResoultionResults {
+  distName: string | null;
+  eventHasDebugIds: boolean;
+  matchingArtifactName: string;
+  projectHasUploadedArtifacts: boolean;
+  releaseName: string | null;
+  releaseSourceMapReference: string | null;
+  sdkDebugIdSupport: 'full' | 'needs-upgrade' | 'unofficial-sdk';
+  sourceFileReleaseNameFetchingResult: 'found' | 'wrong-dist' | 'unsuccessful';
+  sourceFileScrapingStatus:
+    | {status: 'found'}
+    | {error: string; status: 'error'}
+    | {status: 'none'};
+  sourceMapReleaseNameFetchingResult: 'found' | 'wrong-dist' | 'unsuccessful';
+  sourceMapScrapingStatus:
+    | {status: 'found'}
+    | {error: string; status: 'error'}
+    | {status: 'none'};
+  stackFrameDebugId: string | null;
+  stackFramePath: string | null;
+  uploadedSomeArtifact: boolean;
+  uploadedSomeArtifactToRelease: boolean;
+  uploadedSomeArtifactWithDebugId: boolean;
+  uploadedSourceFileWithCorrectDebugId: boolean;
+  uploadedSourceMapWithCorrectDebugId: boolean;
+  sdkVersion?: string;
+}
+
+interface SourceMapsDebuggerModalProps extends ModalRenderProps {
+  sourceResolutionResults: SourceResoultionResults;
+}
+
+export function SourceMapsDebuggerModal({
+  Body,
+  Header,
+  Footer,
+  sourceResolutionResults,
+}: SourceMapsDebuggerModalProps) {
+  const theme = useTheme();
+
+  let debugIdProgress = 0;
+  if (sourceResolutionResults.sdkDebugIdSupport === 'full') {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.stackFrameDebugId !== null) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceFileWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceMapWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  const debugIdProgressPercent = debugIdProgress / 4;
+
+  let releaseNameProgress = 0;
+  if (sourceResolutionResults.releaseName !== null) {
+    releaseNameProgress++;
+  }
+  if (sourceResolutionResults.uploadedSomeArtifactToRelease) {
+    releaseNameProgress++;
+  }
+  if (sourceResolutionResults.sourceFileReleaseNameFetchingResult === 'found') {
+    releaseNameProgress++;
+  }
+  if (sourceResolutionResults.sourceMapReleaseNameFetchingResult === 'found') {
+    releaseNameProgress++;
+  }
+  const releaseNameProgressPercent = releaseNameProgress / 4;
+
+  let scrapingProgress = 0;
+  if (sourceResolutionResults.sourceFileScrapingStatus.status === 'found') {
+    scrapingProgress++;
+  }
+  if (sourceResolutionResults.sourceMapScrapingStatus.status === 'found') {
+    scrapingProgress += 4;
+  }
+  const scrapingProgressPercent = scrapingProgress / 5;
+
+  const [activeTab, setActiveTab] = useState<'debug-ids' | 'release' | 'fetching'>(() => {
```

Comment:
m: can we make active tab state part of url via query params?

Reply:
I thought about this but it doesn't make any sense, as the modal is not active by default on the page. Opening it when query params are there is also not an option as the modal needs knowledge over which exact stack frame it was opened for and stack frames may vary with issue event links - or worded differently, a link to an issue may yield two different stack traces, depending on how it's grouped.

So I think I'll leave this out for now.